system-test/read-rows.ts

```diff
@@ -123,97 +130,89 @@ describe('Bigtable/Table', () => {
-          );
-        }
-        if (reqOpts.rowsLimit) {
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          (requestOptions as any).rowsLimit = reqOpts.rowsLimit;
-        }
-        requestedOptions.push(requestOptions);
-        rowKeysRead.push([]);
-        const requestStream = new PassThrough({objectMode: true});
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (requestStream as any).abort = () => {};
-        dispatch(requestStream, responses!.shift());
-        return requestStream;
+      bigtable = new Bigtable({
+        apiEndpoint: `localhost:${port}`,
       });
-    });
-
-    afterEach(() => {
-      clock.restore();
-      stub.restore();
+      table = bigtable.instance('fake-instance').table('fake-table');
+      service = new BigtableClientMockService(server);
     });
 
     tests.forEach(test => {
-      it(test.name, () => {
-        responses = test.responses;
-        TABLE.maxRetries = test.max_retries;
-        TABLE.createReadStream(test.createReadStream_options)
-          .on('data', row => rowKeysRead[rowKeysRead.length - 1].push(row.id))
-          .on('end', () => (endCalled = true))
-          .on('error', err => (error = err as ServiceError));
-        clock.runAll();
-
-        if (test.error) {
-          assert(!endCalled, ".on('end') should not have been invoked");
-          assert.strictEqual(error!.code, test.error);
-        } else {
-          assert(endCalled, ".on('end') shoud have been invoked");
-          assert.ifError(error);
+      it(test.name, done => {
+        // These variables store request/response data capturing data sent
+        // and received when using readRows with retries. This data is evaluated
+        // in checkResults at the end of the test for correctness.
+        const requestedOptions: google.bigtable.v2.IRowSet[] = [];
+        const responses = test.responses;
+        const rowKeysRead: any[] = [];
+        let endCalled = false;
+        let error: ServiceError | null = null;
+        function checkResults() {
+          if (test.error) {
+            assert(!endCalled, ".on('end') should not have been invoked");
+            assert.strictEqual(error!.code, test.error);
+          } else {
+            assert(endCalled, ".on('end') should have been invoked");
+            assert.ifError(error);
+          }
+          assert.deepStrictEqual(rowKeysRead, test.row_keys_read);
+          assert.strictEqual(
+            responses.length,
+            0,
+            'not all the responses were used'
+          );
+          assert.deepStrictEqual(requestedOptions, test.request_options);
+          done();
         }
-        assert.deepStrictEqual(rowKeysRead, test.row_keys_read);
-        assert.strictEqual(
-          responses.length,
-          0,
-          'not all the responses were used'
-        );
-        assert.deepStrictEqual(requestedOptions, test.request_options);
```

Comment:
These asserts are moved into `checkResults` which evaluates response/request data when the call to createreadstream results in an error or ended stream.